src/Apropos/Gen/Value.hs

```diff
@@ -0,0 +1,41 @@
+{- |
+Module: Apropos.Gen.Value
+Description: Plutus value generators.
+Maintainer: jack@mlabs.city
+
+`Gen`s for `Plutus.V1.Ledger.Value` types.
+-}
+module Apropos.Gen.Value (
+  value,
+  currencySymbol,
+  tokenName,
+) where
+
+import Apropos.Gen (Gen, linear)
+import Apropos.Gen.Api (builtinByteString)
+import Apropos.Gen.Extra (integer)
+import Apropos.Gen.Extra qualified as Gen (map)
+import Plutus.V1.Ledger.Value (
+  CurrencySymbol (CurrencySymbol),
+  TokenName (TokenName),
+  Value (Value),
+ )
+
+-- | Generator for a Plutus `Value`.
+value :: Gen Value
+value = do
+  let tnvMap = Gen.map (linear 1 5) tokenName $ integer (linear 1 maxBound)
+  v <- Gen.map (linear 1 5) currencySymbol tnvMap
+  return $ Value v
+
+-- | Generator for a Plutus `CurrencySymbol`.
+currencySymbol :: Gen CurrencySymbol
```

Comment:
I think it might be better to include some function `ada :: Range -> Gen Value` and another `valueWithAda :: Gen Value` which is the same as `value :: Gen Value` but includes some amount of Ada. Thoughts?

Reply:
It would be nice to have a model for `SingletonValue` that is parameterisable by tokens in a protocol. e.g.

```Haskell
data SingletonValueProp protocolToken =
    IsAda
  | IsProtocolToken protocolToken
  | IsOtherToken
```
Then this can be abstracted into a generator for `MultiValue` that is paramaterised by these value properties.

```Haskell
data MultiValue protocolToken =
  MultiValue (SingletonValue protocolToken)
```

Reply:
Sorry for re-visiting this after so long but I am not sure what you mean @delicious-lemon. I have added the described `ada` and `valueWithAda` functions already. Could we move this into a separate issue?